### PR TITLE
Update Column Infos

### LINKED ISSUE #287: Replace dummy uploaded datasts lists with actual data

### PR DESCRIPTION
**Pull request by dvchristianbors**
Closes #287 (datavisyn/reprovisyn)

Update visyn view column labels to be in line with the Ranking View's column information.
Ranking view column descriptions:
image
visyn view columns
image

Related PR: https://github.com/datavisyn/reprovisyn/pull/310

### CODE REVIEW

File: src/app/workbench/sidebar/CreateNextWorkbenchSidebar.tsx [hunk 1]
```diff
@@ -116,6 +128,7 @@ export function CreateNextWorkbenchSidebar({ workbench }: ICreateNextWorkbenchSi
                         columnDescs: [],
                         data: {},
                         entityId: relationList[0].targetEntity,
+                        formatting: { idColumn: 'id' },
```
Comment (thinkh) [inline src/app/workbench/sidebar/CreateNextWorkbenchSidebar.tsx:131]: Can we always ensure that there is a column with the name `id`?

File: src/store/ordinoSlice.ts [hunk 2]
```diff
@@ -83,6 +83,14 @@ export interface IWorkbench {
    */
   selection: IRow['_visyn_id'][];
 
+  formatting?: {
```
Comment (thinkh) [inline src/store/ordinoSlice.ts:86]: Please add a comment that describes where/for what the formatting is currently used?

File: src/store/ordinoSlice.ts [hunk 3]
```diff
@@ -83,6 +83,14 @@ export interface IWorkbench {
...
+    titleColumn?: string;
+    idColumn: string;
```
Comment (thinkh) [inline src/store/ordinoSlice.ts:88]: I assume these are the column names from the reprovisyn config?

File: src/store/ordinoSlice.ts [hunk 4]
```diff
@@ -83,6 +83,14 @@ export interface IWorkbench {
...
+    formatResult?: string;
```
Comment (thinkh) [inline src/store/ordinoSlice.ts:89]: What does this mean? 🤔 Is it a template string? Please add a comment and add an example.

File: src/store/ordinoSlice.ts [hunk 5]
```diff
@@ -83,6 +83,14 @@ export interface IWorkbench {
...
+    tokenSeparatorsRegex?: string;
+    defaultTokenSeparator?: string;
```
Comment (thinkh) [inline src/store/ordinoSlice.ts:91]: Why are they necessary? Which string is tokenized?

File: src/views/interfaces.ts [hunk 6]
```diff
@@ -50,6 +51,11 @@ export type OrdinoVisynViewPluginType<
       },
       data: IScoreRow<any>[],
     ): void;
+    /**
+     *
+     * @param formatting TODO add typings once Ollie's interface refactoring is finished
```
Comment (thinkh) [inline src/views/interfaces.ts:56]: You can already describe the function in the comment. 😉

File: src/app/workbench/sidebar/DetailsSidebar.tsx [hunk 7]
```diff
@@ -34,7 +34,15 @@ export function DetailsSidebar({ workbench }: IDetailsSidebarProps) {
...
+      const concatStr = ', ';
+      const prevFormatting = ordino.workbenches[workbench.index - 1].formatting;
+      if (prevFormatting) {
+        const selectionDataRow = ordino.workbenches[workbench.index - 1].data[s][prevFormatting.titleColumn || prevFormatting.idColumn];
+        currString += selectionDataRow + concatStr;
+      } else {
+        currString += s + concatStr;
+      }
+      console.log(currString);
     });
```
Comment (thinkh) [inline src/app/workbench/sidebar/DetailsSidebar.tsx:46]:
```ts
const prevWorkbench = ordino.workbenches[workbench.index - 1];
const prevFormatting = prevWorkbench.formatting;
const currString = prevWorkbench.selection.map((selectedId) => {
  return (prevFormatting) ? prevWorkbench.data[selectedId][prevFormatting.titleColumn || prevFormatting.idColumn] : selectedId
}).join(', ');
```

This should do the same as your code. Untested.

* Extract `concatStr` and `prevFormatting` from the `forEach` as it remains the same for each iteration
* Store long repetitive strings such as `ordino.workbenches[workbench.index - 1]` in variables to improve readability
* If you want to concatinate items of an array use `.join()`
* Remove `console.log()`

File: src/app/workbench/sidebar/CreateNextWorkbenchSidebar.tsx [hunk 8]
```diff
@@ -64,11 +64,23 @@ export function CreateNextWorkbenchSidebar({ workbench }: ICreateNextWorkbenchSi
...
+        // if the column data is empty, use id string
+        if (selectionDataRow) {
+          currString += selectionDataRow + concatStr;
+        } else {
+          currString += s + concatStr;
+        }
+      } else {
+        // by default, use the selection string
+        currString += s + concatStr;
+      }
```
Comment (thinkh) [inline src/app/workbench/sidebar/CreateNextWorkbenchSidebar.tsx:79]: Same refactoring applys here.

**Review by thinkh — CHANGES_REQUESTED** [review]
Thanks for your changes. I have a few suggestions for refactorings and mainly missing comments on the interfaces.

File: src/store/ordinoSlice.ts (hunk 2 above)
Comment (thinkh) [reply src/store/ordinoSlice.ts:86]: I saw now in https://github.com/datavisyn/reprovisyn/pull/310/files#diff-20822a6e7462d4b6c46d91c29bdb47cbae71b25d8a2acdfc84547db4fb63d077R77 that this should be of type `IEntityFormatting` but we need to wait for @oltionchampari refactoring, right?

File: src/app/workbench/sidebar/CreateNextWorkbenchSidebar.tsx (hunk 1 above)
Comment (dvchristianbors) [reply src/app/workbench/sidebar/CreateNextWorkbenchSidebar.tsx:131]: woops, did not mean to commit this.

Comment (thinkh) [reply src/app/workbench/sidebar/CreateNextWorkbenchSidebar.tsx:131]: That's why we do code reviews. 😉

Comment (dvchristianbors) [reply src/app/workbench/sidebar/CreateNextWorkbenchSidebar.tsx:131]: thanks for the catch :-)

**Review by thinkh — APPROVED** [review]
Thanks for addressing my feedback. Looks good to me now. 👍

**Pull request merged**